foundation/blockchain/state/state_test.go

```diff
@@ -0,0 +1,332 @@
+	value  uint64
+	tip    uint64
+	data   []byte
+}
+
+// createSignedTX - will create 1 signed transaction.
+func createSignedTX(opts signedTxOpts) (database.SignedTx, error) {
+
+	privateKey, err := crypto.HexToECDSA(opts.keyHex)
+	if err != nil {
+		return database.SignedTx{}, err
+	}
+
+	toAccount, err := database.ToAccountID(opts.to)
+	if err != nil {
+		return database.SignedTx{}, err
+	}
+
+	const chainID = 1
+	tx, err := database.NewTx(chainID, opts.nonce, toAccount, opts.value, opts.tip, opts.data)
+	if err != nil {
+		return database.SignedTx{}, err
+	}
+
+	return tx.Sign(privateKey)
+
+}
+
+// newGenesis will create a new Genesis to be used later.
+func newGenesis() genesis.Genesis {
+	minergenesis := genesis.Genesis{
+		Date:          time.Now().Add(time.Hour * 24 * -365),
+		ChainID:       1,
+		TransPerBlock: 10,
+		Difficulty:    1,
+		MiningReward:  700,
+		GasPrice:      15,
+		Balances: map[string]uint64{
+			"0xF01813E4B85e178A83e29B8E7bF26BD830a25f32": 1000000,
+			"0xdd6B972ffcc631a62CAE1BB9d80b7ff429c8ebA4": 1000000,
+		},
+	}
+
+	return minergenesis
+}
+
+// newTxFactory will create a method that will provide us database.SignedTx
+// with random value at each call, but preserving control over nonce.
+func newTxFactory(t *testing.T) func() database.SignedTx {
+	nonce := uint64(0)
+	mx := sync.Mutex{}
+	s1 := rand.NewSource(time.Now().UnixNano())
+	r1 := rand.New(s1)
+	ret := func() database.SignedTx {
+		mx.Lock()
+		defer mx.Unlock()
+		nonce++
+		so := signedTxOpts{
+			keyHex: PERSONA1_ECDSA,
+			nonce:  nonce,
+			to:     "0xbEE6ACE826eC3DE1B6349888B9151B92522F7F76",
+			value:  r1.Uint64(),
+			tip:    0,
+			data:   []byte{},
+		}
+		ret, err := createSignedTX(so)
+		if err != nil {
+			t.Errorf("Error creating database transaction: %s", err.Error())
+			t.FailNow()
+			return database.SignedTx{}
+		}
+		return ret
+	}
+
+	return ret
+}
+
+// newMiner will create an in memory miner.
+func newMiner(t *testing.T, strkey string) *state.State {
+	if strkey == "" {
+		t.Fatalf("please provide a string w an ECDSA key as strkey")
+	}
+	var err error
+
+	storage, err := memory.New()
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	key, err := crypto.HexToECDSA(MINER1_ECDSA)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	evts := events.New()
+
+	ev := func(v string, args ...any) {
+		const websocketPrefix = "viewer:"
+
+		s := fmt.Sprintf(v, args...)
+		if strings.HasPrefix(s, websocketPrefix) {
+			evts.Send(s)
+		}
+	}
+
+	ret, err := state.New(state.Config{
+		BeneficiaryID:  database.PublicKeyToAccountID(key.PublicKey),
+		Host:           "http://localhost:9080",
+		Genesis:        newGenesis(),
+		Storage:        storage,
+		SelectStrategy: "Tip",
+		KnownPeers:     peer.NewPeerSet(),
+		EvHandler:      ev,
+	})
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	ret.Worker = &noopWorker{}
+	return ret
+}
+
+// ============================ TESTS CASES ===================================
```

Comment:
Let's move the actual tests to the top and all the support below. That is what you really want to see first.

Reply:
Fixed.